requirements/dev.in

```diff
@@ -2,17 +2,18 @@
 -e .
 
 # others
-pylint
-pylint-flask
-tox
-flake8
-flask-testing
-pep8
-pydocstyle
-coverage
-yapf
-isort
-ipdb
-black
-pip-tools
-pytest
+pylint==2.12.2
```

Comment:
How come we lock down the versions in the ".in" file? I haven't used `pip-compile` in my own projects, but I thought that the idea behind a "requirements.in" file was that up-to-date versions would get resolved for us when [`pip-compile` compiles it](https://pypi.org/project/pip-tools/) into a new "requirements.txt" file.

Reply:
I am using pip-compile for the first time as well. My general thought is to have main dependencies with specific versions so that component suddenly doesn't break due to upgrade of a dependency. I thought we could use this to lock versions of transitive dependencies.